src/main/java/de/tum/in/www1/artemis/web/rest/LectureResource.java

```diff
@@ -247,8 +247,7 @@ public ResponseEntity<Lecture> importLecture(@PathVariable long sourceLectureId,
 
         final var savedLecture = lectureImportService.importLecture(sourceLecture, destinationCourse);
 
-        String channelName = generateChannelNameFromTitle(savedLecture.getTitle());
-        Channel createdChannel = channelService.createLectureChannel(savedLecture, channelName);
+        Channel createdChannel = channelService.createLectureChannel(savedLecture, null);
```

Comment:
personal preference: passing optionals instead of null but feel free to ignore as I don't think we have an official guideline for that.

Reply:
I agree, this would also improve the code in the method:
`channelName.orElse(generateChannelNameFromTitle("exercise-", exercise.getTitle())`

Reply:
I agree. I hate nulls

Reply:
Alternative: Provide an overload in `ChannelService`
```java
public Channel createLectureChannel(Lecture lecture) {
     String channelName = generateChannelNameFromTitle("lecture-", lecture.getTitle());
     return createLectureChannel(lecture, channelName);
}
```

That way neither `null` nor `Optional`s needed.

Reply:
I call `createLectureChannel ` in multiple places and I cannot guarantee that the second input is not null. Instead of using overloads I would rather provide an optional here in the function argument.